src/core/namespace.c

```diff
@@ -1812,6 +1812,37 @@ static int create_symlinks_from_tuples(const char *root, char **strv_symlinks) {
         return 0;
 }
 
+static void mount_entry_path_debug_string(const char *root, MountEntry *m, char **error_path) {
+        assert(m);
+
+        /* Create a string suitable for debugging logs, stripping for example the local working directory.
+         * For example, with a BindPaths=/var/bar that does not exist on the host:
+         *
+         * Before:
+         *  foo.service: Failed to set up mount namespacing: /run/systemd/unit-root/var/bar: No such file or directory
+         * After:
+         *  foo.service: Failed to set up mount namespacing: /var/bar: No such file or directory
+         *
+         * Note that this is an error path, so no OOM check is done on purpose. */
+
+        if (!error_path || !mount_entry_path(m))
+                return;
+
+        if (!root) {
+                *error_path = strdup(mount_entry_path(m));
```

Comment:
this is the same line as the last one in the function. hence maybe invert the if check here, and put the next block in the if block and reuse the final line?